pkg/cloud/services/elb/loadbalancer.go

```diff
@@ -169,14 +182,54 @@ func (s *Service) RegisterInstanceWithAPIServerELB(i *infrav1.Instance) error {
 	return nil
 }
 
-// GenerateELBName generates a formatted ELB name
-func GenerateELBName(clusterName string, elbName string) string {
-	return fmt.Sprintf("%s-%s", clusterName, elbName)
+func GenerateELBName(clusterName string, elbName string) (string, error) {
+	standardELBName := generateStandardELBName(clusterName, elbName)
+	if len(standardELBName) <= 32 {
+		return standardELBName, nil
+	}
+	elbName, err := generateHashedELBName(clusterName, "k8s")
+	if err != nil {
+		return "", err
+	}
+	return elbName, nil
+}
+
+// generateStandardELBName generates a formatted ELB name based on cluster
+// and ELB name
+func generateStandardELBName(clusterName string, elbName string) string {
```

Comment:
Would it make sense to change `elbName` to `suffix`?

Reply:
Actually, I'd vote for removing this argument, and pulling `infrav1.APIServerRoleTagValue` down into this method as a hard-coded value.